src/mongo/editors/MongoFindOneResultEditor.ts

```diff
@@ -38,10 +38,11 @@ export class MongoFindOneResultEditor implements ICosmosEditor<IMongoDocument> {
         if (node) {
             result = await (<IAzureNode<MongoDocumentTreeItem>>node).treeItem.update(newDocument);
             node.refresh();
+        } else {
+            // If the node isn't cached already, just update it to Mongo directly (without worrying about updating the tree)
+            const db = await this._databaseNode.treeItem.getDb();
+            result = await MongoDocumentTreeItem.update(db.collection(this._collectionName), newDocument);
```

Comment:
Felt this control flow looked clearer. If you prefer a return statement after each block (functionally the same), I'll do that.

Reply:
I'm fine with either one, really depends on the rest of the code as to which is easier to read/maintain (e.g. how many control flows, how big the then case is, etc.)